test/other/benchmarkMemPoolTypes.cpp

```diff
@@ -0,0 +1,87 @@
+#include "System/MemPoolTypes.h"
+#include "System/Log/ILog.h"
+
+#include <benchmark/benchmark.h>
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+template <size_t T>
+struct ArrayData {
+	char Data[T-8];
+	size_t idx=0;
+};
+
+template <typename TMempool>
+static void BenchStaticMemPoolAllocation(benchmark::State& state) {
+	static TMempool mempool;
+	mempool.clear();
+
+	constexpr size_t page_size = mempool.PAGE_SIZE();
+	using AD = ArrayData<page_size>;
+	std::vector<AD*> allocated;
+
+	size_t idx =0;
+	for (auto _ : state) {
+		if (!mempool.can_alloc()) {
+			state.PauseTiming();
+			mempool.clear();
+			allocated.clear();
+			state.ResumeTiming();
+		}
+		auto* obj = mempool.template alloc<AD>();
+		obj->idx = ++idx;
+		benchmark::DoNotOptimize(obj);
+		allocated.push_back(obj);
+		benchmark::ClobberMemory();
+	}
+}
+
+
+BENCHMARK(BenchStaticMemPoolAllocation<StaticMemPool<10240, 512>>);
+BENCHMARK(BenchStaticMemPoolAllocation<StaticMemPool<10240, 1024>>);
+BENCHMARK(BenchStaticMemPoolAllocation<StaticMemPool<102400, 512>>);
+BENCHMARK(BenchStaticMemPoolAllocation<StaticMemPool<102400, 1024>>);
+
+BENCHMARK(BenchStaticMemPoolAllocation<FixedDynMemPool<512, 16, 256>>);
+BENCHMARK(BenchStaticMemPoolAllocation<FixedDynMemPool<1024, 32, 1024>>);
+BENCHMARK(BenchStaticMemPoolAllocation<FixedDynMemPool<512, 16, 256>>);
+BENCHMARK(BenchStaticMemPoolAllocation<FixedDynMemPool<1024, 32, 1024>>);
+
+template <typename TMempool>
+static void BenchStaticMemPoolAllocationDeallocation(benchmark::State& state) {
+	static TMempool mempool;
+	mempool.clear();
+
+	constexpr size_t page_size = mempool.PAGE_SIZE();
+	using AD = ArrayData<page_size>;
+	std::vector<AD*> allocated;
+
+	size_t idx =0;
+	for (auto _ : state) {
+		if (!mempool.can_alloc()) {
+			for (auto* obj : allocated) {
+				mempool.free(obj);
+			}
+			allocated.clear();
+		}
+		auto* obj = mempool.template alloc<AD>();
+		obj->idx = ++idx;
+		benchmark::DoNotOptimize(obj);
+		allocated.push_back(obj);
+		benchmark::ClobberMemory();
+	}
+}
+
+BENCHMARK(BenchStaticMemPoolAllocationDeallocation<StaticMemPool<10240, 512>>);
```

Comment:
Any specific reasons for the values in the following lines, do they correspond to something salient? Consider leaving a comment like
```
/* 512 is just above the size of struct XYZ.
 * 10240 is a common size of an XYZ Lua table if a game keeps two numbers per unit def, up to NNN defs.
 * 1024000 is an arbitrary check to compare with the above with a higher order of magnitude.
 * 16 is the ... */
 ```

Reply:
fixed in [cb0dd9a](https://github.com/beyond-all-reason/spring/pull/921/commits/cb0dd9ad8b0d9c12fccf102616d97db1e0ed2f42)